src/btree/bt_slvg.c

```diff
@@ -676,7 +709,37 @@ __slvg_trk_leaf(WT_SESSION_IMPL *session, const WT_PAGE_HEADER *dsk, uint8_t *ad
     case WT_PAGE_ROW_LEAF:
         WT_TIME_AGGREGATE_INIT_MERGE(&trk->trk_ta);
         WT_CELL_FOREACH_KV (session, dsk, unpack) {
-            WT_TIME_AGGREGATE_UPDATE(session, &trk->trk_ta, &unpack.tw);
+            switch (unpack.type) {
+            case WT_CELL_KEY:
+                key_size = (int64_t)unpack.size;
+                break;
+            case WT_CELL_KEY_OVFL:
+                key_size = unpack.ovfl_ps.byte_count;
```

Comment:
Here I think I did it wrong. We don't write byte count for key overflow items. To make this work, we have to write the byte count for key overflow items as well.

Reply:
I looked at the code again and I don't think we need to aggregate the counts in salvage. Basically, what salvage does is to find any usable leaf page and connects that to a new internal page to build a new btree. If the leaf page has the aggregation already, we don't need to aggregate that again. If it doesn't, we don't need to aggregate that either, just still leave it as invalid until next time it is reconciled. There is also a code path in salvage that `__slvg_row_build_leaf ` merges keys that are usable from multiple leaf pages. In this code path, we do a reconciliation that should automatically aggregate the counts for us. For the internal pages, they are completely rebuilt in memory and will be reconciled later. Therefore, I think there is no need for us to handle aggregation in salvage.

@mm-ng @bvpvamsikrishna

Reply:
I think I would prefer to write the byte count also for overflow keys. Since overflow key should be rare, it should not generate much extra disk usage. The simplification of the logic is much valuable I think.